File: 1/1.js
const fs = require("fs");

function getData(file) {
  let rawData = fs.readFileSync(file, { encoding: "utf-8" });
  return rawData
    .trim()
    .split("\n")
    .map((el) => parseInt(el));
}

function countIncreases(data, windowSize = 1) {
  let [count, prevSum, currSum] = [0, 0, 0];
  for (let i = 0; i < windowSize; i++) {
    prevSum += data[i];
  }

  for (let i = 0; i < data.length - windowSize; i++) {
    currSum = prevSum - data[i] + data[i + windowSize];
    if (currSum > prevSum) count++;
    prevSum = currSum;
  }
  return count;
}

function main() {
  const data = getData("input.txt");
  let res1 = countIncreases(data);
  let res2 = countIncreases(data, 3);

  console.log(`Part 1: ${res1}`);
  console.log(`Part 2: ${res2}`);
}

main();

File: 1/1.py
def get_data(file: str) -> list[int]:
    with open(file) as file:
        data = [int(i) for i in file.readlines()]

    return data


def count_increases(data: list[int], windowSize: int = 1) -> int:
    count = 0
    prev_sum = sum(data[:windowSize])

    for i in range(len(data) - windowSize):
        curr_sum = prev_sum - data[i] + data[i + windowSize]
        if curr_sum > prev_sum:
            count += 1
        prev_sum = curr_sum

    return count


if __name__ == "__main__":
    data = get_data("input.txt")
    res1 = count_increases(data)
    res2 = count_increases(data, 3)

    print(f"Part 1: {res1}")
    print(f"Part 2: {res2}")
